File: Android/00-TwoActivity/app/src/main/java/com/example/swtichbeetwen2activity/SecondActivity.java
package com.example.swtichbeetwen2activity;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

import org.w3c.dom.Text;

public class SecondActivity extends AppCompatActivity {
    EditText editTextS;
    String stringS;
    Button buttonS;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_second);

        editTextS = findViewById(R.id.editText2);
        buttonS= findViewById(R.id.button2);

        stringS = getIntent().getExtras().getString("Value");
        editTextS.setText(stringS);

        buttonS.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent();
                stringS = editTextS.getText().toString();
                intent.putExtra("Value", stringS);

                setResult(RESULT_OK, intent);
                finish();
            }
        });
    }
}

File: Android/04-Kotlin/app/src/main/java/com/example/kotlinno2/SecondActivity.kt
package com.example.kotlinno2

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.text.Editable
import android.widget.EditText
import androidx.appcompat.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_second.*

class SecondActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_second)
        val value = intent.getStringExtra("Value")

        val nametxt = findViewById<EditText>(R.id.editText2)
        //nametxt.text = Editable.Factory.getInstance().newEditable(value)

        nametxt.setText(value)

        //editText2.setText(intent.getStringExtra("Value"))
        //editText2.text = Editable.Factory.getInstance().newEditable(
        //    value
        //)
        button2?.setOnClickListener {
            val intent = Intent()
            intent.putExtra("Value", editText2.text.toString())

            setResult(Activity.RESULT_OK, intent)
            finish()
        }
    }
}
